components/ui/widget.tsx

```diff
@@ -46,3 +47,25 @@ export const PriceExchange = (props: { from: string; to: string; value: number }
     </span>
   </div>
 );
+
+export const Loading = (props: { text: string }): JSX.Element => (
+  <div className="h-20 flex items-center justify-center text-lg text-white text-opacity-25">
+    <span className="mr-4">{props.text}</span>
+    <span className="scale-150">{Spinner}</span>
+  </div>
+);
+
+export const Transacting = (): JSX.Element => (
+  <div className="border-accent border bg-accent bg-opacity-10 border-solid absolute top-0 right-0 mr-4 mt-4 rounded-full py-1 px-2 text-accent">
+    Transacting {Spinner}
```

Comment:
Common lingo is to say "transaction pending" or "waiting for transaction..." etc.